golem-test-framework/src/model.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use golem_common::model::plugin::{DefaultPluginScope, PluginTypeSpecificDefinition};
+
+#[derive(Debug, Clone)]
+pub struct PluginDefinitionCreation {
```

Comment:
Why do we need a test-framework specific type for this?

Reply:
This serves as an abstraction over the grpc and rest api creation of plugins.

The functions using this used to use the domain type directly, but this no longer works as there is now an extra id / conversion process happening https://github.com/golemcloud/golem/pull/1527/files/d53bd412a1c31aa46e938680dfdb5ee78ae74819#diff-f6c708d1157caac855950eb82eb62277e56663669630375743f5c720497f7da8L1411.

Instead of trying to generalize these types to be usable in both contexts I decided it would be clearer to specialize them. What we have now is:

```
(this thing) -----> rest dto type ----> component service domain type
             \                       /
              \---> grpc dto type - /
```